### PR TITLE
Add `unlisten` function

### LINKED ISSUE #1639: Need Polymer.Base.unlisten to remove the event listener
Currently we have `Polymer.Base.listen` to add an event listener e.g. `tap`.  But we also need `Polymer.Base.unlisten` to remove the event listener.

### PR DESCRIPTION
**Pull request by azakus**
Keep generated event handlers, attach with WeakMap to host element
Add tests for events

Fixes #1639

### CODE REVIEW

File: src/standard/events.html [hunk 1]
```diff
@@ -84,9 +84,34 @@
...
+      var hbl = host.__boundListeners;
+      if (!hbl) {
+        hbl = host.__boundListeners = new WeakMap();
+      }
+      var bl = hbl.get(target);
+      if (!bl) {
+        bl = {};
+        hbl.set(target, bl);
+      }
+      bl[eventName] = handler;
```
Comment (sorvell) [inline src/standard/events.html]:
```
var be = bl[eventName];
if (!be) {
  be = {};
}
be[methodName] = handler;
```

File: src/standard/events.html [hunk 2]
```diff
@@ -84,9 +84,34 @@
...
+    },
+
+    _recallEventHandler: function(host, eventName, target) {
+      var hbl = host.__boundListeners;
+      if (!hbl) {
+        return;
+      }
+      var bl = hbl.get(target);
+      if (!bl) {
```
Comment (sorvell) [inline src/standard/events.html]:
```
var be;
return bl && (be = bl[eventName]) && be[methodName];
```

**Comment by azakus** [comment]
Not sure if `be` is necessary. Since both `methodName` and `eventName` are strings, we can just concatenate them with into one key in `bl`.

**Comment by azakus** [comment]
PTAL @sorvell, multidispatch is supported now.

**Comment by kevinpschaaf** [comment]
LGTUs

**Pull request merged**